packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -94,6 +95,126 @@ function findRoutesAndCreateOptimalPath(
         return fills[0];
     };
 
+    const createPathFromStrategy = (sourcesRustRoute: Float64Array, sourcesOutputAmounts: Float64Array) => {
```

Comment:
This was basically just moved up and wrapped into a function so that we can call it twice (with access to closure variables), first for all sources, then for VIP sources

All of this will need to be rewritten when we rip out the JS router and refactor the fills/path constructs so it's probably fine to leave this a bit hacky solution here for now 🤔